datasetUtils/checkForNewVersion_test.go

```diff
@@ -0,0 +1,102 @@
+package datasetUtils
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"bytes"
+	"log"
+)
+
+func TestFetchLatestVersion(t *testing.T) {
+	// Create a mock HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Return a JSON response similar to the GitHub API
+		w.Write([]byte(`{"tag_name": "v1.0.0"}`))
+	}))
+	defer server.Close()
+	
+	// Replace GitHubAPI with the URL of the mock server
+	oldGitHubAPI := GitHubAPI
+	GitHubAPI = server.URL
+	defer func() { GitHubAPI = oldGitHubAPI }()
+	
+	// Create a mock HTTP client
+	client := server.Client()
+	
+	// Call fetchLatestVersion
+	version, err := fetchLatestVersion(client)
+	if err != nil {
+		t.Fatalf("Expected no error, got %v", err)
+	}
+	
+	// Check the version number
+	if version != "v1.0.0" {
+		t.Errorf("Expected version v1.0.0, got %s", version)
+	}
+}
+
+func TestCheckForNewVersion(t *testing.T) {
+	tests := []struct {
+		name           string
+		currentVersion string
+		mockResponse   string
+		expectedLog    string
+		}{
+			{
+				name:           "New version available",
+				currentVersion: "0.9.0",
+				mockResponse:   `{"tag_name": "1.0.0"}`,
+				expectedLog:    "You should upgrade to a newer version",
+			},
+			{
+				name:           "No new version available",
+				currentVersion: "1.0.0",
+				mockResponse:   `{"tag_name": "1.0.0"}`,
+				expectedLog:    "Your version of this program is up-to-date",
+			},
+		}
+		
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a mock HTTP server
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				// Return a JSON response similar to the GitHub API
+				w.Write([]byte(tt.mockResponse))
+			}))
+			defer server.Close()
+			
+			// Replace GitHubAPI with the URL of the mock server
+			oldGitHubAPI := GitHubAPI
+			GitHubAPI = server.URL
+			defer func() { GitHubAPI = oldGitHubAPI }()
+			
+			// Create a mock HTTP client
+			client := server.Client()
+			
+			// Call CheckForNewVersion
+			CheckForNewVersion(client, "test", tt.currentVersion, false)
+			
+			// Check the log output
+			logOutput := getLogOutput()
+			if !strings.Contains(logOutput, tt.expectedLog) {
```

Comment:
very minor comment, why is this a `contain` rather than `equal`?

Reply:
Because the test is interested in whether `tt.expectedLog` appears anywhere in the `logOutput`, rather than checking if `logOutput` is exactly equal to `tt.expectedLog`.

Reply:
ah ok so logOutput concatenates all the prints in one string?

Reply:
Yes, that's correct. It is capturing all the log messages that were written during the execution of the test case.